core/src/test/scala/org/apache/spark/sql/delta/DeleteSuiteBase.scala

```diff
@@ -504,17 +504,14 @@ abstract class DeleteSuiteBase extends QueryTest
           "v",
           "key >= 1 and value < 3"
         )
-        var result = expectResult
-        checkAnswer(
-          spark.read.format("delta").table("v"),
-          result)
+        checkAnswer(spark.read.format("delta").table("v"), expectResult)
       }
     }
   }
 
   testComplexTempViews("nontrivial projection")(
     text = "SELECT value as key, key as value FROM tab",
-    expectResult = Nil
+    expectResult = Seq(Row(3, 0))
```

Comment:
why did this change??

Reply:
it seems that there was a bug in Delta Lake. to be honest, I'm not entirely sure how my PR fixes that, but if you read this test case carefully it is clear that `Seq(Row(3, 0))` is the correct answer.